suite/tests/linux/syscall_pwait.c

```diff
@@ -49,13 +49,42 @@
 #include <unistd.h>
 #include <time.h>
 #include <poll.h>
+#include <sched.h>
+
+#if defined(X86)
+/* Hardware (-and compiler) read/write memory barrier */
+#    define mb() asm volatile("mfence" : : : "memory")
+/* Hardware (-and compiler) write memory barrier */
+#    define wmb() asm volatile("sfence" : : : "memory")
+/* Hardware (-and compiler) read memory barrier */
+#    define rmb() asm volatile("lfence" : : : "memory")
+#elif defined(AARCH64)
+#    define dsb(opt) asm volatile("dsb " #    opt : : : "memory")
+/* Hardware (-and compiler) read/write memory barrier */
+#    define mb() dsb(sy)
+/* Hardware (-and compiler) read memory barrier */
+#    define rmb() dsb(ld)
+/* Hardware (-and compiler) write memory barrier */
+#    define wmb() dsb(st)
+#else
+#    define dsb(opt) asm volatile("dsb " #    opt : : : "memory")
+/* Hardware (-and compiler) read/write memory barrier */
+#    define mb() dsb(sy)
+/* Hardware (-and compiler) read memory barrier */
+#    define rmb() dsb(sy)
+/* Hardware (-and compiler) write memory barrier */
+#    define wmb() dsb(sy)
+#endif
```

Comment:
I thought we had agreed to use higher-level abstractions and not roll our own synchronization?

Reply:
Ditto above.